packages/smooth_app/lib/widgets/smooth_product_carousel.dart

```diff
@@ -70,9 +83,18 @@ class _SmoothProductCarouselState extends State<SmoothProductCarousel> {
         height: widget.height,
         enableInfiniteScroll: false,
         onPageChanged: (int index, CarouselPageChangedReason reason) {
-          if (index > 0 && InheritedDataManager.of(context).showSearchCard) {
+          _lastIndex = index;
+          if (InheritedDataManager.of(context).showSearchCard) {
```

Comment:
Please use a final variable for `InheritedDataManager.of(context)`, that is used 2 times.